### DISCUSSION TITLE
R code: Detect whether run from quarto

### DISCUSSION THREAD

**Discussion by salim-b** [quarto-dev/quarto-cli#1020] [Q&A] [answered]
What is the recommended way to detect whether or not R code is run as part of a Quarto rendering? Is there a specific environment variable that Quarto sets when it's running?

**Comment by jjallaire** [comment, marked as answer]
The following global knit options are set during a Quarto render: https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/rmd/execute.R#L206-L211

**Reply by salim-b** [reply]
Ok, thank you!

**Reply by dmurdoch** [reply]
I think the code has changed, and those lines no longer give global knit options.  Could you update, or copy the lines?  I think it is these lines, currently 271-277:
```
opts_knit <- list(
    quarto.version = 1,
    rmarkdown.pandoc.from = format$pandoc$from,
    rmarkdown.pandoc.to = to,
    rmarkdown.version = 3,
    rmarkdown.runtime = "static"
  )
```

**Reply by salim-b** [reply]
@dmurdoch Yup, because the link wasn't a permalink.

Original permalink: https://github.com/quarto-dev/quarto-cli/blob/b7878d22688aa60c243862ddbf77805d36dc8f34/src/resources/rmd/execute.R#L206-L212

Current permalink: https://github.com/quarto-dev/quarto-cli/blob/a604268ea7724d8991512f1f7ed89ce7b84f391a/src/resources/rmd/execute.R#L271-L277
